includes/Modules/Analytics_4.php

```diff
@@ -0,0 +1,308 @@
+				'label' => __( 'Analytics 4 snippet placed', 'google-site-kit' ),
+				'value' => $settings['useSnippet'] ? __( 'Yes', 'google-site-kit' ) : __( 'No', 'google-site-kit' ),
+				'debug' => $settings['useSnippet'] ? 'yes' : 'no',
+			),
+		);
+	}
+
+	/**
+	 * Gets map of datapoint to definition data for each.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Map of datapoints to their definitions.
+	 */
+	protected function get_datapoint_definitions() {
+		return array(
+			'GET:accounts'              => array( 'service' => 'analyticsadmin' ),
+			'GET:properties'            => array( 'service' => 'analyticsadmin' ),
+			'GET:webdatastreams'        => array( 'service' => 'analyticsadmin' ),
+			'POST:create-property'      => array(
+				'service'                => 'analyticsadmin',
+				'scopes'                 => array( 'https://www.googleapis.com/auth/analytics.edit' ),
+				'request_scopes_message' => __( 'You’ll need to grant Site Kit permission to create a new Analytics 4 property on your behalf.', 'google-site-kit' ),
+			),
+			'POST:create-webdatastream' => array(
+				'service'                => 'analyticsadmin',
+				'scopes'                 => array( 'https://www.googleapis.com/auth/analytics.edit' ),
+				'request_scopes_message' => __( 'You’ll need to grant Site Kit permission to create a new Analytics 4 web data stream on your behalf.', 'google-site-kit' ),
+			),
+		);
+	}
+
+	/**
+	 * Creates a request object for the given datapoint.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request $data Data request object.
+	 * @return RequestInterface|callable|WP_Error Request object or callable on success, or WP_Error on failure.
+	 *
+	 * @throws Invalid_Datapoint_Exception Thrown if the datapoint does not exist.
+	 */
+	protected function create_data_request( Data_Request $data ) {
+		switch ( "{$data->method}:{$data->datapoint}" ) {
+			case 'GET:accounts':
+				return $this->get_service( 'analyticsadmin' )->accounts->listAccounts();
+			case 'GET:properties':
+				if ( ! isset( $data['accountID'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'accountID' ),
+						array( 'status' => 400 )
+					);
+				}
+				return $this->get_service( 'analyticsadmin' )->properties->listProperties(
+					array(
+						'filter' => 'parent:accounts/' . $data['accountID'],
+					)
+				);
+			case 'GET:webdatastreams':
+				if ( ! isset( $data['propertyID'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'propertyID' ),
+						array( 'status' => 400 )
+					);
+				}
+				return $this->get_service( 'analyticsadmin' )->properties_webDataStreams->listPropertiesWebDataStreams( 'properties/' . $data['propertyID'] );
+			case 'POST:create-property':
```

Comment:
We currently sort the cases by datapoint first, then by method. So the `create-*` datapoints would come after `accounts` and before `properties`.